Changing the text on the 2021 State of InnerSource page

clcoffey opened the pull request:
I have changed the text on this page to reflect the fact that we are currently working on the report.

Is there any chance we could remove the Read Research Report button for the moment until the report is published?

@spier is there any chance you can help with removing the button again?

dicortazar commented:
Thanks a lot @claredillon , this works for me :)

clcoffey committed:
Changing the text on the 2021 State of InnerSource page

I have changed the text on this page to reflect the fact that we are currently working on the report.

Is there any chance we could remove the Read Research Report button for the moment, until the report is published?

clcoffey committed:
Update state-of-innersource-survey-2021.md

spier commented:
I can try.

To confirm, the research pages needs 3 different states now:

1. survey is active - show "Take Survey" button
2. survey is closed but report isn't finished yet - show no button
3. report is finished - show "Read Research Report"  button

Correct?

clcoffey commented:
Yes, that is correct. Thanks Sebastian!

spier commented:
Hmm. I don't really know how to do this somewhat simple using `hugo`.
I suspect that with a static site generator like that, it is always somewhat difficult (maybe impossible) to make the pages look differently depending on which date it is.

So if anybody else has an idea here how to do this, please share :)

voborgus commented:
@spier as I see the possible solution is to use custom page parameters with dates for research pages and compare them using `.After` or `.Before`.

For example, each research page should have:
```
closing_date: 2022-12-12
finish_date: 2022-12-30
```

The [layout](https://github.com/InnerSourceCommons/innersourcecommons.org/blob/master/layouts/research/single.html) should have something like:
```
{{ if and (now.Before .Params.finish_date) (now.After .Params.closing_date) }}
...
{{ else if ... }}
...
{{ }}
```

[Documentation](https://gohugohq.com/howto/compare-date-strings-in-hugo/)

clcoffey committed:
Changing text on buttons

I tried my best but I am getting errors I don't know how to fix.

Initially it was telling me that it expected time.Time and was getting string, and I kept trying to convert string to time.

Now it is saying that can't give argument to non-function now.Before .Params.finish_date

clcoffey commented:
@voborgus is there any chance you can have a look at this and tell me what I am doing wrong?

clcoffey committed:
half fix

voborgus committed:
string to time function fix

spier commented on layouts/research/list.html:
```diff
@@ -14,7 +14,13 @@
             <!-- <h3><a href="{{ .Permalink }}" class="post-title">{{ .Title }}</a></h3> -->
             <p class="card-text">{{ .Summary }}</p>
-            <a href="{{.Permalink}}" class="btn btn-primary btn-sm">{{ cond (eq .Params.survey_active true) "Take Survey" "Read Research Report" }}</a>
+            {{ if and (now.Before (time .Params.finish_date)) (now.After (time .Params.closing_date)) }}
```
I didn't try this out. Just doing it in my head now.
So please try it before using it :)

An alternative to writing this logic would be:

```html
{{ if (now.Before (time .Params.closing_date)) }}
  <a href="{{.Params.research_url}}" target="_blank" class="btn btn-primary">Take Survey</a>
{{ else }}
  {{ if (now.Before (time .Params.finish_date)) }}
    <a href="{{.Params.research_url}}" target="_blank" class="btn btn-primary">Coming Soon</a>
  {{ else }}
    <a href="{{.Params.research_url}}" target="_blank" class="btn btn-primary">Read Research Report</a>
  {{ end }}
{{ end }}
```

While this is more nesting of the if/else clauses, I like it because:
* the different branches are in the same order that the report will go through i.e. first "Take Survey", then "Coming Soon", then "Read Research Report".
* it only uses `now.Before` (and not `now.After`)

spier reviewed (commented):
I didn't check if this actually works.
We don't have test cases for this and I wouldn't know how to create them in Hugo either.

However in theory this should work :)

My main comments are:

* understanding what `closing_date` and `finish_date` means will be hard to remember when creating a new report, as both sound so similar. maybe slightly longer variable names would help like `survey_closing_date` and `report_finish_date`?
* additionally I wonder if there is a way to do this where we would not have to update the markdown file for each stage that the survey is in? (e.g. `state-of-innersource-survey-2021.md`)

In either case, none of this needs to stop the PR from merging. Just ideas for possible improvements.

Nicely done!

spier reviewed (approved):
In addition to the comments in my previous review, I am approving the PR so that you can merge the PR if you like.

clcoffey committed:
Changed variable names

clcoffey commented:
Thanks Sebastian. I think you are right about the variable names and I made the change.
